src/java.desktop/share/classes/javax/swing/JEditorPane.java

```diff
@@ -472,69 +472,82 @@ public void setPage(URL page) throws IOException {
         if (page == null) {
             throw new IOException("invalid url");
         }
-        URL loaded = getPage();
 
+        final URL loaded = getPage();
+        final Object postData = getPostData();
+        final String reference = page.getRef();
 
-        // reset scrollbar
-        if (!page.equals(loaded) && page.getRef() == null) {
-            scrollRectToVisible(new Rectangle(0,0,1,1));
-        }
-        boolean reloaded = false;
-        Object postData = getPostData();
-        if ((loaded == null) || !loaded.sameFile(page) || (postData != null)) {
-            // different url or POST method, load the new content
-
-            int p = getAsynchronousLoadPriority(getDocument());
-            if (p < 0) {
-                // open stream synchronously
-                InputStream in = getStream(page);
-                if (kit != null) {
-                    Document doc = initializeModel(kit, page);
-
-                    // At this point, one could either load up the model with no
-                    // view notifications slowing it down (i.e. best synchronous
-                    // behavior) or set the model and start to feed it on a separate
-                    // thread (best asynchronous behavior).
-                    p = getAsynchronousLoadPriority(doc);
-                    if (p >= 0) {
-                        // load asynchronously
-                        setDocument(doc);
-                        synchronized(this) {
-                            pageLoader = new PageLoader(doc, in, loaded, page);
-                            pageLoader.execute();
-                        }
-                        return;
-                    }
-                    read(in, doc);
-                    setDocument(doc);
-                    reloaded = true;
-                }
+        if ((postData == null) && page.sameFile(loaded)) {
```

Comment:
This line of code causes lots of test failures because `page.sameFile(loaded)` doesn't accept a `null` parameter, which results in `NullPointerException`.